frontend/src/features/admin-form/create/builder-and-design/components/BuilderAndDesignDrawer/EditFieldDrawer/EditRadio.tsx

```diff
@@ -0,0 +1,200 @@
+import { useMutateFormFields } from '../../../mutations'
+import { isPendingFormField } from '../../../utils'
+
+import { DrawerContentContainer } from './DrawerContentContainer'
+import { FormFieldDrawerActions } from './FormFieldDrawerActions'
+
+export interface IEditRadioProps {
+  field: RadioFieldSchema
+}
+
+interface IEditRadioInputs
+  extends Pick<
+    RadioFieldBase,
+    'title' | 'description' | 'required' | 'othersRadioButton'
+  > {
+  fieldOptions: string
+}
+
+const transformRadioOpts = {
+  toArray: (input?: string) =>
+    input
+      ?.split('\n')
+      .map((opt) => opt.trim())
+      .filter(Boolean) ?? [],
+  toString: (output?: string[]) => output?.filter(Boolean).join('\n'),
+}
+
+const transformToFormField = ({
+  fieldOptions,
+  ...rest
+}: IEditRadioInputs): Partial<RadioFieldSchema> => {
+  return {
+    ...rest,
+    fieldOptions: transformRadioOpts.toArray(fieldOptions),
+  }
+}
+
+const requiredValidationRule = createBaseValidationRules({ required: true })
+
+const fieldOptionsValidationRule = {
+  ...requiredValidationRule,
+  validate: {
+    duplicate: (opts: string) => {
+      const optsArray = transformRadioOpts.toArray(opts)
+      return (
+        new Set(optsArray).size === optsArray.length ||
+        'Please remove duplicate options.'
+      )
+    },
+  },
+}
+
+export const EditRadio = ({ field }: IEditRadioProps): JSX.Element => {
+  const { updateActiveField, clearActiveField } = useEditFieldStore()
+
+  const {
+    handleSubmit,
+    reset,
+    register,
+    watch,
+    formState: { errors, isDirty, isSubmitting },
+  } = useForm<IEditRadioInputs>({
+    defaultValues: {
+      title: field.title,
+      description: field.description,
+      fieldOptions: transformRadioOpts.toString(field.fieldOptions),
+      required: field.required,
+      othersRadioButton: field.othersRadioButton,
+    },
+  })
+
+  const watchedInputs = watch()
+
+  useDebounce(
+    () => {
+      if (!watchedInputs) return
+      return updateActiveField(transformToFormField(watchedInputs))
+    },
+    300,
+    [
+      transformRadioOpts,
+      // Required destructure to prevent debounce firing infinitely.
+      watchedInputs.description,
+      watchedInputs.fieldOptions,
+      watchedInputs.required,
+      watchedInputs.title,
+      watchedInputs.othersRadioButton,
+    ],
+  )
+
+  const { mutateFormField } = useMutateFormFields()
+
+  const saveButtonText = useMemo(
+    () => (isPendingFormField(field) ? 'Create' : 'Save'),
+    [field],
+  )
+
+  const isSaveDisabled = useMemo(
+    () => isDirty || isPendingFormField(field),
+    [field, isDirty],
+  )
+
+  const handleUpdateField = handleSubmit((inputs) => {
+    const updatedField = extend({}, field, transformToFormField(inputs))
+    return mutateFormField.mutate(updatedField, {
+      onSuccess: () => {
+        reset(inputs)
+      },
+    })
+  })
+
+  const [tabIndex, setTabIndex] = useState(0)
```

Comment:
I think the new design have no more tabs, can update and remove all the tab logic liow ✨